pkgs/development/libraries/geos/default.nix

```diff
@@ -1,22 +1,33 @@
-{ lib, stdenv, fetchurl }:
+{ lib
+, stdenv
+, fetchurl
+, cmake }:
 
 stdenv.mkDerivation rec {
   pname = "geos";
-  version = "3.9.1";
+  version = "3.10.2";
 
   src = fetchurl {
     url = "https://download.osgeo.org/geos/${pname}-${version}.tar.bz2";
-    sha256 = "sha256-fmMFB9ysncB1ZdJJom8GoVyfWwxS3SkSmg49OB1+OCo=";
+    sha256 = "sha256-ULvFmaw4a0wrOWLcxBHwBAph8gSq7066ciXs3Qz0VxU=";
   };
 
+  buildInputs = [ cmake ];
+
   enableParallelBuilding = true;
 
-  # https://trac.osgeo.org/geos/ticket/993
-  configureFlags = lib.optional stdenv.isAarch32 "--disable-inline";
+  postPatch = ''
+    substituteInPlace tools/geos-config.in \
+      --replace "@libdir@" "@prefix@/lib" \
```

Comment:
maybe its still worth investigating what actually sets `@libdir@` and `@includedir@` to wrong values
maybe that will fix the other issues `valhalla`
if i get some free time today I'll have a look at `valhalla`, but don't want to block this from getting merged

Reply:
I got as far as noting that valhalla is getting stuck with 3.10.x for brew users with https://github.com/valhalla/valhalla/issues/3388, but it doesn't seem totally resolved with them either, leading me to figure the best case was to just leave valhalla with 3.9 and let the rest move forward for now.

Reply:
If, as noted in https://github.com/valhalla/valhalla/issues/3388 and https://github.com/libgeos/geos/issues/500, then https://github.com/libgeos/geos/issues/543 will be reflected with 3.11.0, so I'd consider this the best resolution for 3.10, and geos39 can be removed when geos bumps to 3.11.